modules/ding_event/ding_event.module

```diff
@@ -538,3 +598,54 @@ function ding_event_preprocess_node(&$variables) {
     }
   }
 }
+
+/**
+ * Implements hook_views_post_execute().
+ */
+function ding_event_views_post_execute(&$view) {
+  if ($view->name == 'ding_event' && $view->current_display != 'event_libraries_list') {
+    $nids = array_map(function ($node) {
+      return $node->nid;
+    }, $view->result);
+    $nodes = node_load_multiple($nids);
+    $view->result = array_slice(_ding_event_group($nodes), 0, $view->items_per_page);
+  }
+}
+
+/**
+ * Get events in different groups.
+ */
+function _ding_event_group(array $nodes) {
+  $events[] = reset($nodes);
+  foreach ($nodes as $node) {
+    if (!_ding_event_same_group($events, $node)) {
+      $events[] = $node;
+    }
+  }
+
+  return $events;
+}
+
+/**
+ * Check if events are in the same group.
+ *
+ * Same group means they have the same title, date and library.
+ */
+function _ding_event_same_group($nodes, $event) {
+  $group = FALSE;
+  foreach ($nodes as $node) {
+    $event_date = $event->field_ding_event_date[LANGUAGE_NONE][0]['value'] ? $event->field_ding_event_date[LANGUAGE_NONE][0]['value'] : '';
+    $event_date = new DateTime($event_date);
+    $node_date = $node->field_ding_event_date[LANGUAGE_NONE][0]['value'] ? $node->field_ding_event_date[LANGUAGE_NONE][0]['value'] : '';
+    $node_date = new DateTime($node_date);
+
+    if ($event->title == $node->title &&
+      $event_date->diff($node_date)->d <= 1 &&
+      $event->og_group_ref[LANGUAGE_NONE][0]['target_id'] == $node->og_group_ref[LANGUAGE_NONE][0]['target_id']) {
+      $group = TRUE;
+      break;
+    }
+  }
+
+  return $group;
+}
```

Comment:
This diff is to render one event instead of multiple, when their start date is within one day, have same title and are attached to same library.

Reply:
![image](https://user-images.githubusercontent.com/574498/100233197-10b37a00-2f32-11eb-8cd1-d6c05525d1a4.png)
vs
![image](https://user-images.githubusercontent.com/574498/100233315-3e98be80-2f32-11eb-927f-578c12079f68.png)

Reply:
Great functionality. Let's port this!